src/luascript.cpp

```diff
@@ -9391,55 +9391,52 @@ int LuaScriptInterface::luaPlayerHasMount(lua_State* L) {
 	return 1;
 }
 
-int LuaScriptInterface::luaPlayerGetPremiumDays(lua_State* L)
+int LuaScriptInterface::luaPlayerGetPremiumTime(lua_State* L)
 {
-	// player:getPremiumDays()
+	// player:getPremiumTime()
 	Player* player = getUserdata<Player>(L, 1);
 	if (player) {
-		lua_pushnumber(L, player->premiumDays);
+		lua_pushnumber(L, std::max<int32_t>(0, player->premiumEndsAt - time(nullptr)));
 	} else {
 		lua_pushnil(L);
 	}
 	return 1;
 }
 
-int LuaScriptInterface::luaPlayerAddPremiumDays(lua_State* L)
+int LuaScriptInterface::luaPlayerAddPremiumTime(lua_State* L)
 {
-	// player:addPremiumDays(days)
+	// player:addPremiumTime(seconds)
 	Player* player = getUserdata<Player>(L, 1);
 	if (!player) {
 		lua_pushnil(L);
 		return 1;
 	}
 
-	if (player->premiumDays != std::numeric_limits<uint16_t>::max()) {
-		uint16_t days = getNumber<uint16_t>(L, 2);
-		int32_t addDays = std::min<int32_t>(0xFFFE - player->premiumDays, days);
-		if (addDays > 0) {
-			player->setPremiumDays(player->premiumDays + addDays);
-			IOLoginData::addPremiumDays(player->getAccount(), addDays);
-		}
-	}
+	uint32_t seconds = getNumber<uint32_t>(L, 2);
+	int32_t oldTime = std::max<int32_t>(0, player->premiumEndsAt - time(nullptr));
+	player->setPremiumTime(time(nullptr) + oldTime + seconds);
+	IOLoginData::updatePremiumTime(player->getAccount(), player->premiumEndsAt);
```

Comment:
Why calculate time left from now when it's not used?

```suggestion
	uint32_t seconds = getNumber<uint32_t>(L, 2);
	player->setPremiumTime(player->premiumEndsAt + seconds);
	IOLoginData::updatePremiumTime(player->getAccount(), player->premiumEndsAt);
```